include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,329 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan Core 1.0 */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
+    //uint32_t              maxPushConstantsSize;
+    //uint32_t              maxMemoryAllocationCount;
+    //uint32_t              maxSamplerAllocationCount;
+    //VkDeviceSize          bufferImageGranularity;
+    //VkDeviceSize          sparseAddressSpaceSize;
```

Comment:
expose
https://github.com/Devsh-Graphics-Programming/Nabla/pull/348#discussion_r867399121

Reply:
if we want to expose `sparseAddressSpaceSize` should we expose `VkPhysicalDeviceSparseProperties`?
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceSparseProperties.html

Reply:
`bufferImageGranularity` is related to alignment of linear (like buffer) and non-linear resourse(like image) in the same VkDeviceMemory.
This happens in the case of suballocation where we could have multiple resources in a DeviceMemory,
Obviously doesn't make sense in OpenGL so this could be 1u in GL.

Reply:
yes the content of `vkPhysicalDeviceSparseProperties` should go here

Reply:
```
 //uint32_t              maxMemoryAllocationCount;
 //uint32_t              maxSamplerAllocationCount;
```
what should I report for these on OpenGL?
I didn't find anything specifying any Max for buffer objects or sampler objects in opengl

Reply:
there's no such thing in GL, just report 1M samplers and 1G allocations